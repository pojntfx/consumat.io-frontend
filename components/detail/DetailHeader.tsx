import { Media } from "../../lib/api/consumat-io";
import { imageSizes, useImage } from "../../hooks/ImageHook";
import MediaImage from "../helper/MediaImage";
import FavoriteButton from "../helper/FavoriteButton";

type DetailHeaderProps = {
  media: Media;
};

const DetailHeader = ({ media }: DetailHeaderProps) => {
  return (
    <div>
      <div
        className="flex items-end relative pl-52 bg-gray-500 w-full h-64 sm:h-96 -mt-4 rounded-b shadow-md"
        style={{
          backgroundImage: `linear-gradient(0deg, rgba(31, 41, 55, 0.8), rgba(31, 41, 55, 0.8)), url(${useImage(
            imageSizes.backdrop.w1280,
            media.backdropPath
          )})`,
          backgroundSize: "cover",
        }}
      >
        <FavoriteButton className="absolute right-4 top-4" media={media} />
        <h2
          className="text-white text-2xl sm:text-4xl md:text-6xl px-0 sm:px-2 pb-4 max-w-sm md:max-w-2xl"
          style={{ textShadow: "2px 2px 2px rgba(31, 41, 55)" }}
        >
          {media.title}
        </h2>
      </div>
      <div className="px-8">
        <div className="relative w-40">
          <span
            className={`flex items-center justify-center text-white font-semibold absolute w-10 h-10 -top-3 -right-3 rounded-full shadow ${
              media.ratingAverage < 5
                ? "bg-red-500"
                : media.ratingAverage < 6.5
                ? "bg-yellow-500"
                : media.ratingAverage < 8
                ? "bg-green-500"
                : "bg-green-700"
            }`}
          >
            {media.ratingAverage}
          </span>
          <div className="w-40 h-60 -mt-48 overflow-hidden rounded self-center shadow-md md:flex-shrink-0">
            <MediaImage
              className="w-40 h-60"
              imageSrc={useImage(imageSizes.poster.w500, media.posterPath)}
            />
          </div>
        </div>
      </div>
    </div>
  );
};

export default DetailHeader;
