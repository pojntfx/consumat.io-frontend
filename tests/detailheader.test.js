import React from "react";
import renderer from "react-test-renderer";
import DetailHeader from "../components/detail/DetailHeader";
import testExamples from "./testExamples";
import { MockedProvider } from "@apollo/client/testing";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <MockedProvider mocks={testExamples.exampleMocks} addTypename={false}>
        <DetailHeader media={testExamples.exampleMedia} />
      </MockedProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
