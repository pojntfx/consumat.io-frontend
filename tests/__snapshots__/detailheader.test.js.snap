// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="flex items-end relative pl-52 bg-gray-500 w-full h-64 sm:h-96 -mt-4 rounded-b shadow-md"
    style={
      Object {
        "backgroundColor": "rgba(31, 41, 55, 0.8)",
        "backgroundImage": "linear-gradient(0deg, rgba(31, 41, 55, 0.8), rgba(31, 41, 55, 0.8))undefined)",
        "backgroundSize": "cover",
      }
    }
  >
    <button
      className="bg-transparent focus:outline-none shadow-none hover:shadow-none absolute right-4 top-4"
      onClick={[Function]}
    >
      <svg
        className="h-8 w-8 text-red-500 shadow-none hover:shadow-none hover:fill-current false undefined"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4.318 6.318a4.5 4.5 0 000 6.364L12 20.364l7.682-7.682a4.5 4.5 0 00-6.364-6.364L12 7.636l-1.318-1.318a4.5 4.5 0 00-6.364 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={2}
        />
      </svg>
    </button>
    <h2
      className="text-white text-2xl sm:text-4xl md:text-6xl px-0 sm:px-2 pb-4 max-w-sm md:max-w-2xl"
      style={
        Object {
          "textShadow": "2px 2px 2px rgba(31, 41, 55)",
        }
      }
    />
  </div>
  <div
    className="px-8"
  >
    <div
      className="relative w-40"
    >
      <span
        className="flex items-center justify-center text-white font-semibold absolute w-10 h-10 -top-3 -right-3 rounded-full shadow bg-green-700"
      >
        -
      </span>
      <div
        className="w-40 h-60 -mt-48 overflow-hidden rounded self-center shadow-md md:flex-shrink-0"
      >
        <div
          className="w-40 h-60 flex-shrink-0"
        >
          <img
            className="h-full w-full object-cover"
            onError={[Function]}
            src={null}
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
